### PR TITLE
stylix: set GTK icon theme

### PR DESCRIPTION
**Pull request by laycookie**
Closes: https://github.com/danth/stylix/issues/458

### CODE REVIEW

File: stylix/hm/icon.nix
Comment (trueNAHO) [inline stylix/hm/icon.nix:10]: Is this a debugging leftover or is this meant to be removed?

**Review by trueNAHO — COMMENTED** [review]
Is this PR ready for review? I will mark this PR as draft for organizational reasons. Feel free to mark this PR as ready when it is.

Make sure to address the GitHub warnings when looking at https://github.com/danth/stylix/pull/603/files.

File: stylix/hm/icon.nix
Comment (laycookie) [reply stylix/hm/icon.nix]: Ya this was a leftover from me testing some stuff before, removing it in the next commit.

**Comment by laycookie** [comment]
I resolved all the linting warnings, however I was unable to test this build because I broke my nix distro. (long story) It should still work but if anyone could test my branch that would be ideal.
Note: If you enable iconTheme, and set a `package` option, but not set `light` or `dark` which correspond to the names of the iconTheme for there appropriate polarity, the config will fail to build. This is intended behavior which was inherited directly from gtk.

File: stylix/hm/icon.nix
Comment (trueNAHO) [inline stylix/hm/icon.nix:18]: This is very clever. However, it might be better to prioritize readability with simple if-statements.

File: stylix/hm/icon.nix
Comment (trueNAHO) [inline stylix/hm/icon.nix:8]: Could you clarify why the `pkgs.stdenv.hostPlatform.isLinux` guard is used? If it is actually necessary, consider explaining it in a code comment just above.

**Review by trueNAHO — CHANGES_REQUESTED** [review]

File: stylix/hm/icon.nix
Comment (laycookie) [reply stylix/hm/icon.nix:8]: I believe some people use nix configs on macos, and I don't think you can change icons on mac in this way. I don't have mac to test it out so just for safety I'm restricting this option to only work on Linux, so people could easily use the same config across both devices without modified it on each device.

File: stylix/hm/icon.nix
Comment (laycookie) [reply stylix/hm/icon.nix:18]: Initially I did this as an if statement in my head, but it came out very large and fairly confusing. This code does require some knowledge about functional programing, but it comes out much more simpler if you do know them.
For reference this statement unfolds in to 3 if statements, one of which nests an additional if statement. If you insist I can attempt to rewrite it though.

File: stylix/hm/icon.nix
Comment (trueNAHO) [reply stylix/hm/icon.nix:8]: > I believe some people use nix configs on macos, and I don't think you can change icons on mac in this way. I don't have mac to test it out

Me neither.

> just for safety I'm restricting this option to only work on Linux, so people could easily use the same config across both devices without modified it on each device.

I suggest optimistically removing the guard in this PR and re-adding the guard if someone complains in the future.

Comment (laycookie) [reply stylix/hm/icon.nix:8]: Alright I can do it right now give me 5 minutes.

File: stylix/hm/icon.nix
Comment (trueNAHO) [reply stylix/hm/icon.nix:18]: > For reference this statement unfolds in to 3 if statements, one of which nests an additional if statement.

The nested if-statement, which is probably the most confusing part could be flattened to something like:

```nix
if polarity == "dark"
then dark
else if polarity == "light"
then light
```

> If you insist I can attempt to rewrite it though.

Just send it as a comment here before pushing it. Seeing both makes it easier to compare which one is better.

Comment (laycookie) [reply stylix/hm/icon.nix:18]: Something like this.
```nix
name = if cfg.light != null && cfg.dark != null then
	if polarity == "dark"
	then cfg.dark
	else if polarity == "light"
	then cfg.light
else if cfg.light == null 
then cfg.dark
else if cfg.dark == null 
then cfg.light;
```

Comment (laycookie) [reply stylix/hm/icon.nix:18]: Also I'm not entirely sure how an `else` should look like in here.

Comment (trueNAHO) [reply stylix/hm/icon.nix:18]: Should we not simplify the `name` to:

```nix
name = "stylix";
```

Many Stylix themes are called `Stylix` or `stylix`.

Comment (laycookie) [reply stylix/hm/icon.nix:18]: No because this field is the name of the iconTheme, so based on what is going to be here the iconTheme will be selected. I implemented the same interface as someone in the #458 suggested.

Comment (laycookie) [reply stylix/hm/icon.nix:18]: @trueNAHO Is there any other questions regarding the code, or are we just waiting for danth to approve of this?

Comment (trueNAHO) [reply stylix/hm/icon.nix:18]: > Is there any other questions regarding the code

Considering the scope of this icon implementation it might be good to get @danth's opinion and approval.

> or are we just waiting for danth to approve of this?

Since the current implementation is fine for me, we can just leave it like this and move forward.

**Review by trueNAHO — APPROVED** [review]
```
Reviewed-by: NAHO <90870942+trueNAHO@users.noreply.github.com>
```

**Review by KamWithK — APPROVED** [review]
Tested and works well

**Pull request merged**
